WcaOnRails/spec/controllers/competitions_controller_spec.rb

```diff
@@ -221,6 +221,23 @@
         sign_in organizer
       end
 
+      it 'cannot pass a non-delegate as delegate' do
+        delegate_ids_old = competition.delegate_ids
+        fake_delegate = FactoryGirl.create(:user)
+        post :update, id: competition, competition: { delegate_ids: fake_delegate.id }
+        invalid_competition = assigns(:competition)
+        expect(invalid_competition.errors.messages[:delegate_ids]).to eq ["You must pass only delegates here"]
+        competition.reload
+        expect(competition.delegate_ids).to eq delegate_ids_old
+      end
+
+      it 'can change the delegate' do
+        new_delegate = FactoryGirl.create(:delegate)
+        post :update, id: competition, competition: { delegate_ids: new_delegate.id }
+        competition.reload
+        expect(competition.delegate_ids).to eq new_delegate.id.to_s
```

Comment:
Looks like we overwrite `delegate_ids`: https://github.com/cubing/worldcubeassociation.org/blob/master/WcaOnRails/app/models/competition.rb#L118
ActiveRecord's `has_many` already adds such a method which returns an array, that's what confused me.